plugins/lsp/default.nix

```diff
@@ -1,231 +1,225 @@
-        internal = true;
-        visible = false;
-      };
 
-      inlayHints = mkOption {
-        description = ''
-          Whether to enable LSP inlay-hints.
-          Only affects language servers with inlay-hints support.
+              extraOptions = mkOption {
+                type = attrsOf anything;
+                description = "Extra options for the server";
+              };
+            };
+          })
+        ]);
+      description = "A list of enabled LSP servers. Don't use this directly.";
+      default = [ ];
+      internal = true;
+      visible = false;
+    };
 
-          See [`:h lsp-inlay_hint`](https://neovim.io/doc/user/lsp.html#lsp-inlay_hint).
-        '';
-        type = types.bool;
-        default = false;
-        example = true;
-      };
+    inlayHints = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable LSP inlay-hints.
+        Only affects language servers with inlay-hints support.
 
-      onAttach = mkOption {
-        type = types.lines;
-        description = "A lua function to be run when a new LSP buffer is attached. The argument `client` and `bufnr` is provided.";
-        default = "";
-      };
+        See [`:h lsp-inlay_hint`](https://neovim.io/doc/user/lsp.html#lsp-inlay_hint).
+      '';
+    };
 
-      capabilities = mkOption {
-        type = types.lines;
-        description = "Lua code that modifies inplace the `capabilities` table.";
-        default = "";
-      };
+    onAttach = mkOption {
+      type = types.lines;
+      description = "A lua function to be run when a new LSP buffer is attached. The argument `client` and `bufnr` is provided.";
+      default = "";
+    };
 
-      setupWrappers = mkOption {
-        type = with types; listOf (functionTo str);
-        description = "Code to be run to wrap the setup args. Takes in an argument containing the previous results, and returns a new string of code.";
-        default = [ ];
-      };
+    capabilities = mkOption {
+      type = types.lines;
+      description = "Lua code that modifies inplace the `capabilities` table.";
+      default = "";
+    };
 
-      preConfig = mkOption {
-        type = types.lines;
-        description = "Code to be run before loading the LSP. Useful for requiring plugins";
-        default = "";
-      };
+    setupWrappers = mkOption {
+      type = with types; listOf (functionTo str);
+      description = "Code to be run to wrap the setup args. Takes in an argument containing the previous results, and returns a new string of code.";
+      default = [ ];
+    };
 
-      postConfig = mkOption {
-        type = types.lines;
-        description = "Code to be run after loading the LSP. This is an internal option";
-        default = "";
-      };
+    preConfig = mkOption {
+      type = types.lines;
+      description = "Code to be run before loading the LSP. Useful for requiring plugins";
+      default = "";
```

Comment:
I wonder if `preConfig` and `postConfig` can be rename-alias modules pointing to `lubConfig.pre` and `luaConfig.post` respectively?

Reply:
I think it could, but it would need a little refactor